jetty-server/src/main/java/org/eclipse/jetty/server/session/AbstractSessionDataStore.java

```diff
@@ -114,49 +200,63 @@ public void store(String id, SessionData data) throws Exception
         if (data == null)
             return;
 
-        final AtomicReference<Exception> exception = new AtomicReference<Exception>();
+        final RunnableResult<Object> result = new RunnableResult<>();
 
-        Runnable r = new Runnable()
+        Runnable r = () ->
```

Comment:
This is much better, but the lambda is a little on the big size.  Perhaps move the body to an inner class like:
```java
    private class RunStore extends RunnableResult<Object> implements Runnable
    {
        String _id;
        SessionData data;
        ... etc.
        public void run()
        {
            // do everything from lambda and save in self as a result
        }
    }
```
Then this method becomes:
```java
        final RunStore runStore = new RunStore(id, data);
        _context.run(runStore);
        runStore.throwIfException();
```

Reply:
I've greatly reduced the scope of the Runnable - it really only needs to surround the call to the subclass' doStore() method.